lib/pure/os.nim

```diff
@@ -3217,6 +3184,46 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
+proc sameFileContent*(path1, path2: string; checkSize = false; bufferSize = 8192.Positive): bool {.
+    rtl, extern: "nos$1", tags: [ReadIOEffect], noWeirdTarget.} =
+  ## Returns true if both pathname arguments refer to files with identical
+  ## binary content.
+  ##
+  ## If `checkSize` is `true` then checks the file sizes *before reading the files*,
+  ## if the files have different file sizes they can not have the same contents,
+  ## `checkSize = true` may be faster, specially for very big files.
+  ##
+  ## See also:
+  ## * `sameFile proc <#sameFile,string,string>`_
+  var
+    a, b: File
+  if not open(a, path1): return false
+  if not open(b, path2):
+    close(a)
+    return false
+  if checkSize and getFileInfo(a).size != getFileInfo(b).size:
+    close(a)
+    close(b)
+    return false
+  var bufA = alloc(bufferSize)
+  var bufB = alloc(bufferSize)
+  while true:
+    var readA = readBuffer(a, bufA, bufferSize)
```

Comment:
this is still not correct wrt fd leaks! the only correct way is defer or try/finally; but try/finally adds 2 nested scopes, and defer adds 0 syntactically.

readBuffer can raise `IOError`

Reply:
That code is pre-existent, but makes sense.

Reply:
indeed, it's pre-existant; then either fix it in this PR (would be great) or file an issue so we don't forget

Reply:
oh, can you add a comment in the PR body mentioning you fixed this bug?